src/Context/QueryContext.js

```diff
@@ -0,0 +1,29 @@
+import { createContext, useState, useContext } from "react";
+
+const QueryContext = createContext();
+const QueryUpdateContext = createContext();
+
+export function useQuery() {
+  return useContext(QueryContext);
+}
+
+export function useQueryUpdate() {
+  return useContext(QueryUpdateContext);
+}
+
+export function QueryProvider({children}) {
+  const [query, setQuery] = useState('Wizeline');
+
+  function changeQuery(newQuery) {
+    setQuery(newQuery);
+  }
+
+
+  return (
+  <QueryContext.Provider value={query}>
```

Comment:
Using a single context like this, doesn't work for you?
```suggestion
  <QueryContext.Provider value={{query, changeQuery}}>
```